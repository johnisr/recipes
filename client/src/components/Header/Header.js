import React, { Component } from 'react';
import { connect } from 'react-redux';

export class Header extends Component {
  renderContent() {
    switch (this.props.auth) {
      case null:
        return null;
      case false:
        return (
          <li>
            <a href="/auth/google">Login With Google</a>
          </li>
        );
      default:
        return [
          <li key="1">
            <a href="/auth/logout">Logout</a>
          </li>,
        ];
    }
  }
  render() {
    return (
      <nav>
        <div className="nav-wrapper">
          <ul className="right">{this.renderContent()}</ul>
        </div>
      </nav>
    );
  }
}

const mapStateToProps = state => ({
  auth: state.auth,
});

export default connect(mapStateToProps)(Header);
