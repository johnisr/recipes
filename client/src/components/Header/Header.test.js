import React from 'react';
import { shallow } from 'enzyme';
import { Header } from './Header';

describe('Should render correctly', () => {
  test('When auth has not been fetched', () => {
    const wrapper = shallow(<Header auth={null} />);
    expect(wrapper).toMatchSnapshot();
  });
  test('When auth has been fetched and is true', () => {
    const wrapper = shallow(<Header auth />);
    expect(wrapper).toMatchSnapshot();
  });
  test('When auth has been fetched and is false', () => {
    const wrapper = shallow(<Header auth={false} />);
    expect(wrapper).toMatchSnapshot();
  });
});
