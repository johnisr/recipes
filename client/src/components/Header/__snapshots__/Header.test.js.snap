// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly When auth has been fetched and is false 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    auth={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="nav-wrapper"
>
        <ul
                className="right"
        >
                <li>
                        <a
                                href="/auth/google"
                        >
                                Login With Google
                        </a>
                </li>
        </ul>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className="right"
>
          <li>
                    <a
                              href="/auth/google"
                    >
                              Login With Google
                    </a>
          </li>
</ul>,
        "className": "nav-wrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <li>
            <a
                        href="/auth/google"
            >
                        Login With Google
            </a>
</li>,
          "className": "right",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="/auth/google"
>
              Login With Google
</a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Login With Google",
              "href": "/auth/google",
            },
            "ref": null,
            "rendered": "Login With Google",
            "type": "a",
          },
          "type": "li",
        },
        "type": "ul",
      },
      "type": "div",
    },
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="nav-wrapper"
>
          <ul
                    className="right"
          >
                    <li>
                              <a
                                        href="/auth/google"
                              >
                                        Login With Google
                              </a>
                    </li>
          </ul>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ul
            className="right"
>
            <li>
                        <a
                                    href="/auth/google"
                        >
                                    Login With Google
                        </a>
            </li>
</ul>,
          "className": "nav-wrapper",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <li>
              <a
                            href="/auth/google"
              >
                            Login With Google
              </a>
</li>,
            "className": "right",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                href="/auth/google"
>
                Login With Google
</a>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Login With Google",
                "href": "/auth/google",
              },
              "ref": null,
              "rendered": "Login With Google",
              "type": "a",
            },
            "type": "li",
          },
          "type": "ul",
        },
        "type": "div",
      },
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Should render correctly When auth has been fetched and is true 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    auth={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="nav-wrapper"
>
        <ul
                className="right"
        >
                <li>
                        <a
                                href="/auth/logout"
                        >
                                Logout
                        </a>
                </li>
        </ul>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className="right"
>
          <li>
                    <a
                              href="/auth/logout"
                    >
                              Logout
                    </a>
          </li>
</ul>,
        "className": "nav-wrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <a
                            href="/auth/logout"
              >
                            Logout
              </a>
</li>,
          ],
          "className": "right",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": <a
                href="/auth/logout"
>
                Logout
</a>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Logout",
                "href": "/auth/logout",
              },
              "ref": null,
              "rendered": "Logout",
              "type": "a",
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "div",
    },
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="nav-wrapper"
>
          <ul
                    className="right"
          >
                    <li>
                              <a
                                        href="/auth/logout"
                              >
                                        Logout
                              </a>
                    </li>
          </ul>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ul
            className="right"
>
            <li>
                        <a
                                    href="/auth/logout"
                        >
                                    Logout
                        </a>
            </li>
</ul>,
          "className": "nav-wrapper",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <a
                                href="/auth/logout"
                >
                                Logout
                </a>
</li>,
            ],
            "className": "right",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": <a
                  href="/auth/logout"
>
                  Logout
</a>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Logout",
                  "href": "/auth/logout",
                },
                "ref": null,
                "rendered": "Logout",
                "type": "a",
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
        "type": "div",
      },
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Should render correctly When auth has not been fetched 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    auth={null}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="nav-wrapper"
>
        <ul
                className="right"
        />
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className="right"
/>,
        "className": "nav-wrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": null,
          "className": "right",
        },
        "ref": null,
        "rendered": null,
        "type": "ul",
      },
      "type": "div",
    },
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="nav-wrapper"
>
          <ul
                    className="right"
          />
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ul
            className="right"
/>,
          "className": "nav-wrapper",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": null,
            "className": "right",
          },
          "ref": null,
          "rendered": null,
          "type": "ul",
        },
        "type": "div",
      },
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
